### PR TITLE
Sherlock-33, 112: Unit test to confirm issues are invalid

### PR DESCRIPTION
**Pull request by prat-gpt**
# Description of change
## High level
* See 
  - https://github.com/sherlock-audit/2023-04-ajna-judging/issues/112
  - https://github.com/sherlock-audit/2023-04-ajna-judging/issues/33
* Add test to confirm the above issues are invalid

### CODE REVIEW

**Review by grandizzy — APPROVED** [review]
LGTM, thank you!

**Review by ith-harvey — APPROVED** [review]

**Review by MikeHathaway — APPROVED** [review]
LGTM

File: tests/forge/unit/Rewards/RewardsManager.t.sol [hunk 1]
```diff
@@ -2344,6 +2344,180 @@ contract RewardsManagerTest is RewardsHelperContract {
...
+            from:               _minterOne,
+            tokenId:            tokenIdOne,
+            minAmountToReceive: 0,
+            epochsClaimed:      _epochsClaimedArray(2,0),
+            reward:             totalRewardEarned
+        });
+        
+        // ensure total rewards earned are same with and without unstake between epochs
+        assertEq(_ajnaToken.balanceOf(_minterOne), totalRewardEarned);
+
```
Comment (EdNoepel) [inline tests/forge/unit/Rewards/RewardsManager.t.sol:2518]: nit: extra space here and 2348

**Review by EdNoepel — APPROVED** [review]
Sherlock issue 112 discusses stake/unstake within same block, which this tests nicely.  33, however, discusses stake/unstake within same epoch, which spans many blocks.  While the tests are similar, one could argue blocks should be skipped when testing issue 33.

Regardless, this is good due diligence to prove both issues invalid.

**Pull request merged**